src/applications/mhv-medical-records/reducers/labsAndTests.js

```diff
@@ -142,23 +151,32 @@ const convertChemHemRecord = record => {
  * @returns the appropriate frontend object for display
  */
 const convertMicrobiologyRecord = record => {
+  const specimenRef =
+    isArrayAndHasItems(record.specimen) && record.specimen[0].reference;
+  const collectionRequest = extractContainedResource(record, specimenRef);
```

Comment:
It might be good to pull this out into an `extractSpecimen` function. Then you can easily write unit tests for it. You should be writing unit tests for all new functionality.